src/colmap/sensor/models.h

```diff
@@ -1538,6 +1561,128 @@ void ThinPrismFisheyeCameraModel::Distortion(
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+// RadTanThinPrismFisheyeModel
+
+std::string RadTanThinPrismFisheyeModel::InitializeParamsInfo() {
+  return "fx, fy, cx, cy, k0, k1, k2, k3, k4, k5, p0, p1, s0, s1, s2, s3";
+}
+
+std::array<size_t, 2> RadTanThinPrismFisheyeModel::InitializeFocalLengthIdxs() {
+  return {0, 1};
+}
+
+std::array<size_t, 2>
+RadTanThinPrismFisheyeModel::InitializePrincipalPointIdxs() {
+  return {2, 3};
+}
+
+std::array<size_t, 12>
+RadTanThinPrismFisheyeModel::InitializeExtraParamsIdxs() {
+  return {4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
+}
+
+std::vector<double> RadTanThinPrismFisheyeModel::InitializeParams(
+    const double focal_length, const size_t width, const size_t height) {
+  std::vector<double> params(RadTanThinPrismFisheyeModel::num_params);
+  params[0] = focal_length;
+  params[1] = focal_length;
+  params[2] = width / 2.0;
+  params[3] = height / 2.0;
+
+  for (size_t i = 4; i < RadTanThinPrismFisheyeModel::num_params; ++i) {
+    params[i] = 0;
+  }
+
+  return params;
+}
+
+template <typename T>
+void RadTanThinPrismFisheyeModel::ImgFromCam(
+    const T* params, T u, T v, T w, T* x, T* y) {
+  const T f1 = params[0];
+  const T f2 = params[1];
+  const T c1 = params[2];
+  const T c2 = params[3];
+
+  u /= w;
+  v /= w;
+
+  T du, dv;
+  Distortion(&params[4], u, v, &du, &dv);
+  *x = u + du;
+  *y = v + dv;
+
+  *x = f1 * *x + c1;
+  *y = f2 * *y + c2;
+}
+
+template <typename T>
+void RadTanThinPrismFisheyeModel::CamFromImg(
+    const T* params, const T x, const T y, T* u, T* v, T* w) {
+  const T f1 = params[0];
+  const T f2 = params[1];
+  const T c1 = params[2];
+  const T c2 = params[3];
+
+  *u = (x - c1) / f1;
+  *v = (y - c2) / f2;
+  *w = 1;
+
+  IterativeUndistortion(&params[4], u, v);
+}
+
+template <typename T>
+void RadTanThinPrismFisheyeModel::Distortion(
+    const T* extra_params, const T u, const T v, T* du, T* dv) {
+  const int numK = 6;
+  const T radial_coeffs[numK] = {extra_params[0],
+                                 extra_params[1],
+                                 extra_params[2],
+                                 extra_params[3],
+                                 extra_params[4],
+                                 extra_params[5]};
+
+  const T p0 = extra_params[6];
+  const T p1 = extra_params[7];
+  const T s0 = extra_params[8];
+  const T s1 = extra_params[9];
+  const T s2 = extra_params[10];
+  const T s3 = extra_params[11];
+
+  const T r = ceres::sqrt(u * u + v * v);
+  const T theta = ceres::atan(r);
+  const T theta2 = theta * theta;
+  T th_radial = T(1);
+  T theta_power = T(1);
+  for (int i = 0; i < numK; ++i) {
+    theta_power *= theta2;
+    th_radial += radial_coeffs[i] * theta_power;
+  }
+
+  const T theta_divr =
+      (r < std::numeric_limits<T>::epsilon()) ? static_cast<T>(1.0) : theta / r;
+
+  const T x = th_radial * theta_divr * u;
```

Comment:
Should we move the theta_dive mapping outside the distortion function, such that the distortion returns zero offset for extra params all zero

Reply:
Ok we could also do it a separate PR by also updating OpenCVFisheye. Lets keep it in the current way for now